### ISSUE TITLE
Allow indexes for models to be defined by configuration

### ISSUE THREAD

**Issue by Sammaye** [lordthorzonus/yii2-algolia#21]
As I have been building and testing algolia functions in my site I have noticed that actually I want two indexes per model:

- Production
- Development

and I would like these indexes to be switchable via the applied environments config.

Maybe should allow a indexes config param for the main extension setup which allows for something like:
```
'indexes' => [
    'common\models\Title' => ['titles']
]
```
That way you can overwrite it for dev and prod configurations

**Comment by lordthorzonus** [comment]
Hi, this is already possible :+1:. You can define your environment in the extension config [example here](https://github.com/lordthorzonus/yii2-algolia#using-multiple-environments). So for example your titles index would become `dev_titles` `prod_titles` according to Yii env.

This works if you use any of the methods from the `leinonen\Yii2Algolia\AlgoliaManager` (`pushToIndices`, `updateInIndices`, `removeFromIndices`, reindex stuff etc. ) or the `leinonen\Yii2Algolia\Searchable`.  Here's also a test case that shows how it works [link](https://github.com/lordthorzonus/yii2-algolia/blob/master/tests/Integration/AlgoliaManagerTest.php#L298-L324).

So basically for all indices the env gets prefixed to the name. This is also what [Algolia recommends](https://www.algolia.com/doc/guides/security/multiple-environments/). It allows you also to restrict API keys for certain indices which is quite cool :).

**Comment by Sammaye** [comment]
Ah thanks, I will have to try this :)
